backend/metasfresh-webui-api/src/main/java/de/metas/ui/web/edi_desadv/ChangeEDI_ExportStatusHelper.java

```diff
@@ -0,0 +1,156 @@
+		}
+	}
+
+	public void EDI_DesadvDoIt(@NonNull final EDIDesadvId desadvId, @NonNull final EDIExportStatus targetExportStatus, final boolean isProcessed)
+	{
+		final I_EDI_Desadv edi = desadvDAO.retrieveById(desadvId);
+		edi.setEDI_ExportStatus(targetExportStatus.getCode());
+		edi.setProcessed(isProcessed);
+		desadvDAO.save(edi);
+	}
+
+	public void C_DocOutbound_LogDoIt(final EDIExportStatus targetExportStatus, final ArchiveId logId)
+	{
+		// technical detail: the I_C_Doc_Outbound_Log is updated when we update the C_Invoice, via interceptor: de.metas.edi.model.validator.C_Invoice.updateDocOutBoundLog
+		final de.metas.edi.model.I_C_Doc_Outbound_Log docOutboundLog = ediDocOutBoundLogService.retreiveById(logId);
+
+		final TableRecordReference invoiceRecordReference = TableRecordReference.ofReferenced(docOutboundLog);
+
+		final InvoiceId invoiceId = InvoiceId.ofRepoId(invoiceRecordReference.getRecord_ID());
+
+		ChangeEDI_ExportStatusHelper.C_InvoiceDoIt(invoiceId, targetExportStatus);
+
+		docOutboundLog.setEDI_ExportStatus(targetExportStatus.getCode());
+		ediDocOutBoundLogService.save(docOutboundLog);
+	}
+
+	public void C_InvoiceDoIt(final InvoiceId invoiceId, final EDIExportStatus targetExportStatus)
+	{
+		final de.metas.edi.model.I_C_Invoice invoice = ediDocOutBoundLogService.retreiveById(invoiceId);
+		invoice.setEDI_ExportStatus(targetExportStatus.getCode());
+		invoiceDAO.save(invoice);
+	}
+
+	@NonNull
+	public LookupValuesList computeTargetExportStatusLookupValues(final EDIExportStatus fromExportStatus)
+	{
+		final List<EDIExportStatus> availableTargetStatuses = ChangeEDI_ExportStatusHelper.getAvailableTargetExportStatuses(fromExportStatus);
+
+		return availableTargetStatuses.stream()
+				.map(s -> LookupValue.StringLookupValue.of(s.getCode(), adReferenceDAO.retrieveListNameTranslatableString(EDIExportStatus.AD_Reference_ID, s.getCode())))
+				.collect(LookupValuesList.collect());
+	}
+
+	@Nullable
+	public Object computeParameterDefaultValue(final EDIExportStatus fromExportStatus)
+	{
+		final List<EDIExportStatus> availableTargetStatuses = ChangeEDI_ExportStatusHelper.getAvailableTargetExportStatuses(fromExportStatus);
+		if (!availableTargetStatuses.isEmpty())
+		{
+			final String code = availableTargetStatuses.get(0).getCode();
+			return LookupValue.StringLookupValue.of(code, adReferenceDAO.retrieveListNameTranslatableString(EDIExportStatus.AD_Reference_ID, code));
+		}
+		return IProcessDefaultParametersProvider.DEFAULT_VALUE_NOTAVAILABLE;
+	}
+
+	/**
+	 * TODO tbp: how to check if this docOutboundLog is for edi or not?
+	 * docOutboundLog.isEdiEnabled() is a virtual column!
+	 * in swing this is checked by """(select bp.IsEdiInvoicRecipient from C_BPartner bp where bp.C_BPartner_ID=C_Doc_Outbound_Log.C_BPartner_ID)"""
+	 * do i need to check for if the AD_Table is C_Invoice as well?
+	 * help maybe?
+	 */
+	public boolean checkIsInvoiceAndEDI(final I_C_Doc_Outbound_Log docOutboundLog)
```

Comment:
in theory you could check for C_Invoice.IsEdiEnabled..but that's an extra load..
or you could check for edi-export-status to be not null..but you already do that in the process anyways..

Reply:
Checking for edi-export-status to be not null is not enough, because the `C_Doc_Outbound_Log` might also reference `Sales Order`s or `Delivery Note`s, and we don't want to change their IDs afaiu.

Reply:
> or you could check for edi-export-status to be not null

...in addition to the check for `C_Invoice` which you do already

Reply:
done